### PR TITLE
ts strict part 2: wallets

### PR DESCRIPTION
**Pull request by chrisnojima**
Please see https://github.com/keybase/client/pull/18383 for an overview

### CODE REVIEW

File: shared/actions/wallets.tsx [hunk 1]
```diff
@@ -1192,37 +1254,37 @@ const refreshTrustlinePopularAssets = () =>
...
-    )
-  ).then(() => WalletsGen.createRefreshTrustlineAcceptedAssets({accountID}))
+    ).then(() => WalletsGen.createRefreshTrustlineAcceptedAssets({accountID}))
+  )
 }
 
-const deleteTrustline = (state, {payload: {accountID, assetID}}) => {
-  const asset = state.wallets.trustline.assetMap.get(assetID, Constants.emptyTrustline)
+const deleteTrustline = (state: TypedState, {payload: {accountID, assetID}}) => {
+  const asset = state.wallets.trustline.assetMap.get(assetID, Constants.emptyAssetDescription)
```
Comment (chrisnojima) [inline shared/actions/wallets.tsx]: I believe this was just wrong, please verify

File: shared/actions/wallets.tsx [hunk 2]
```diff
@@ -1192,37 +1254,37 @@ const refreshTrustlinePopularAssets = () =>
     })
   )
 
-const addTrustline = (state, {payload: {accountID, assetID}}) => {
-  const asset = state.wallets.trustline.assetMap.get(assetID, Constants.emptyTrustline)
+const addTrustline = (state: TypedState, {payload: {accountID, assetID}}) => {
+  const asset = state.wallets.trustline.assetMap.get(assetID, Constants.emptyAssetDescription)
```
Comment (chrisnojima) [inline shared/actions/wallets.tsx]: I believe this was just wrong, please verify

File: shared/constants/wallets.tsx [hunk 3]
```diff
@@ -159,15 +159,15 @@ export const makeBuiltPayment = I.Record<Types._BuiltPayment>({
 })
 
 export const makeSEP7Summary = I.Record<Types._SEP7Summary>({
-  fee: null,
+  fee: -1,
```
Comment (chrisnojima) [inline shared/constants/wallets.tsx]: null isn't valid so i just used this

File: shared/wallets/send-form/body/advanced.tsx [hunk 4]
```diff
@@ -34,21 +34,25 @@ const SecretNoteAndPublicMemo = () => {
   const publicMemo = buildingAdvanced.publicMemo.stringValue()
 
   // We currently don't support path payment requests, so no need to check if this is a request
-  const secretNoteMaxLength = Container.useSelector(state => state.wallets.staticConfig.paymentNoteMaxLength)
-  const publicMemoMaxLength = Container.useSelector(state => state.wallets.staticConfig.publicMemoMaxLength)
+  const secretNoteMaxLength = Container.useSelector(state =>
+    state.wallets.staticConfig ? state.wallets.staticConfig.paymentNoteMaxLength : 0
```
Comment (chrisnojima) [inline shared/wallets/send-form/body/advanced.tsx]: config can be null

File: shared/wallets/send-form/note-and-memo/container.tsx [hunk 5]
```diff
@@ -10,9 +10,12 @@ export const SecretNote = namedConnect(
     const recipientType = state.wallets.building.recipientType
     const building = state.wallets.building
     const built = building.isRequest ? state.wallets.builtRequest : state.wallets.builtPayment
-    const maxLength = building.isRequest
-      ? state.wallets.staticConfig.requestNoteMaxLength
-      : state.wallets.staticConfig.paymentNoteMaxLength
+    // TODO is this ok
+    const maxLength = state.wallets.staticConfig
```
Comment (chrisnojima) [inline shared/wallets/send-form/note-and-memo/container.tsx]: config can be null

File: shared/wallets/send-form/note-and-memo/container.tsx [hunk 6]
```diff
@@ -32,7 +35,7 @@ export const PublicMemo = namedConnect(
   state => {
     const building = state.wallets.building
     const built = state.wallets.builtPayment
-    const maxLength = state.wallets.staticConfig.publicMemoMaxLength
+    const maxLength = state.wallets.staticConfig ? state.wallets.staticConfig.publicMemoMaxLength : 0
```
Comment (chrisnojima) [inline shared/wallets/send-form/note-and-memo/container.tsx]: same here

File: shared/wallets/send-form/pick-asset.tsx [hunk 7]
```diff
@@ -77,15 +76,15 @@ const AssetList = ({accountID, isSender, username}) => {
                       type="BodyExtrabold"
                       lineClamp={1}
                       ellipsizeMode="tail"
-                      style={selected && styles.textSelected}
+                      style={selected ? styles.textSelected : undefined}
```
Comment (chrisnojima) [inline shared/wallets/send-form/pick-asset.tsx]: false isn't a valid style (vs collapseStyles)

File: shared/wallets/sep7-confirm/asset-input-container.tsx [hunk 8]
```diff
@@ -12,7 +11,7 @@ const mapStateToProps = (state: Container.TypedState) => {
   const currency = 'XLM'
   return {
     bottomLabel: `Your primary account has ${
-      state.wallets.sep7ConfirmInfo.availableToSendNative
+      state.wallets.sep7ConfirmInfo ? state.wallets.sep7ConfirmInfo.availableToSendNative : '(unknown)' // TODO is this ok
```
Comment (chrisnojima) [inline shared/wallets/sep7-confirm/asset-input-container.tsx]: this seems technically possible

File: shared/wallets/sep7-confirm/container.tsx [hunk 9]
```diff
@@ -19,7 +19,7 @@ const mapDispatchToProps = (dispatch: Container.TypedDispatch) => ({
 })
 
 const Connected = Container.connect(mapStateToProps, mapDispatchToProps, (stateProps, dispatchProps) => {
-  if (stateProps.loading) {
+  if (stateProps.loading || !stateProps.sep7ConfirmInfo /* TODO is this ok */) {
```
Comment (chrisnojima) [inline shared/wallets/sep7-confirm/container.tsx]: i believe this is an implicit guarantee

File: shared/wallets/sep7-confirm/index.tsx [hunk 10]
```diff
@@ -201,14 +201,14 @@ const SEP7Confirm = (props: Props) => (
         {!!props.callbackURL && <CallbackURLBanner callbackURL={props.callbackURL} />}
         {props.operation === 'pay' ? (
           <PaymentInfo
-            amount={props.amount}
+            amount={props.amount || '' /* is this ok? */}
```
Comment (chrisnojima) [inline shared/wallets/sep7-confirm/index.tsx]: can be null actually

File: shared/wallets/sep7-confirm/index.tsx [hunk 11]
```diff
@@ -201,14 +201,14 @@ const SEP7Confirm = (props: Props) => (
...
             availableToSendNative={props.availableToSendNative}
             displayAmountFiat={props.displayAmountFiat}
             memo={props.memoType === 'MEMO_TEXT' ? props.memo : ''}
             message={props.message}
             onChangeAmount={props.onChangeAmount}
-            recipient={props.recipient}
-            userAmount={props.userAmount}
+            recipient={props.recipient || '' /* is this ok? */}
```
Comment (chrisnojima) [inline shared/wallets/sep7-confirm/index.tsx]: here too

File: shared/wallets/sep7-confirm/index.tsx [hunk 12]
```diff
@@ -230,7 +230,7 @@ const SEP7Confirm = (props: Props) => (
           type="Success"
           onClick={
             props.operation === 'pay'
-              ? () => props.onAcceptPay(props.amount || props.userAmount)
+              ? () => props.onAcceptPay(props.amount || props.userAmount || '' /* TODO is this ok? */)
```
Comment (chrisnojima) [inline shared/wallets/sep7-confirm/index.tsx]: here too

File: shared/wallets/transaction-details/index.tsx [hunk 13]
```diff
@@ -322,13 +322,15 @@ const TransactionDetails = (props: NotLoadingProps) => {
...
           ? 'Unknown issuer'
           : Constants.shortenAccountID(props.sourceIssuerAccountID))
+  const {issuerAccountID} = props
   const destinationIssuer =
     props.assetCode === ''
       ? 'Stellar Lumens'
       : props.issuerDescription ||
-        (props.issuerAccountID === Types.noAccountID
+        // TODO is this ok?
+        (issuerAccountID === Types.noAccountID || !issuerAccountID
```
Comment (chrisnojima) [inline shared/wallets/transaction-details/index.tsx]: can also be null

File: shared/wallets/wallet/header/maybe-switcher.tsx [hunk 14]
```diff
@@ -21,6 +21,7 @@ const _Switcher: React.FunctionComponent<Props & Kb.OverlayParentProps> = props
 
 const Switcher = Kb.OverlayParentHOC(_Switcher)
 
+// TODO these types don't make sense
```
Comment (chrisnojima) [inline shared/wallets/wallet/header/maybe-switcher.tsx]: how this works is weird i think

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/wallets/send-form/note-and-memo/container.tsx (hunk 5 above)
Comment (nathunsmitty) [reply shared/wallets/send-form/note-and-memo/container.tsx]: I think this is what we want to do. `staticConfig` should be set by an RPC at load, but in the event it isn't, I think 0 is a fine value to default to.

**Pull request merged**